### ISSUE TITLE
Local DNS server ignored: IP address connection works, hostname connection shaky

### ISSUE THREAD

**Issue by prajjwald** [nextcloud/android#1891]
### Actual behaviour
When attempting to connect to my local network hosted server (lets call it **myserver.localdomain, ip address: 192.168.1.100**): the nextcloud android client displays the following behavior:

- takes a very long time to connect: often says 'unable to connect', occasionally connects
- even when it connects (after multiple attempts to get it to do so), and I login: loading the folders/files takes a very long time.  File downloads fail citing connection issues.
- the server itself has redis caching enabled: if I load the nextcloud on google chrome on the same phone running nextcloud client: everything is snappy.

As soon as I switch to using the ip address (using the hostname/ip address I listed above for the example), I get the following behavior:
https://192.168.1.100/nextcloud - everything works **very well**.  Login is instantaneous, file/directory listings are instantaneous
https://myserver.localdomain/nextcloud - very problematic connection behavior

**As a sidenote: please note that there is no problem in the linux and windows desktop clients, only in the android one.**

### Expected behaviour
My phone and android tablet are both running on the local network, and are using the local router for dns name resolutions.  There should be the exact same behavior for accessing the local (but not public to internet) domain names as for accessing via I.P. address.
 
### Steps to reproduce
1. Host nextcloud on a local server with a made-up domain name (that does not exist in the internet).  Make sure the local dns server (usually wifi router) resolves this name.  In my case: my router resolves <hostname>.<domainname I set in router> for each new host that connects to it.
2. Access it from nextcloud client using the local hostname: problematic behavior
3. Access it from the browser using the same hostname: behavior is fine
4. Access it from nextcloud client using the hardcoded ip address: behavior is fine

### Environment data
Android version:

Device model: 

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by prajjwald** [comment]
Apologies, I think this is because of my local router settings.  I was looking at my wireless router DNS settings today, and I noticed that several older incarnations of my server (different mac addresses) were still around, despite having been removed long ago.  While my desktop or the browser in my phone do not seem to have any problems resolving the host correctly: as soon as I got rid of the lingering duplicate entries: it looks like the client works fine with hostname (as opposed to i.p. address) urls too.
